pkg/gogocodec/gogocodec.go

```diff
@@ -23,6 +23,7 @@ const (
 	jaegerProtoGenPkgPath = "github.com/jaegertracing/jaeger/proto-gen"
 	jaegerModelPkgPath    = "github.com/jaegertracing/jaeger/model"
 	otelProtoPkgPath      = "go.opentelemetry.io/collector"
+	etcdProtoPkgPath      = "go.etcd.io/etcd/api/v3/etcdserverpb"
```

Comment:
nice fix, but i'm wondering if we can just update our etcd dependency instead of adding it here?

does the most recent etcd support latest grpc/proto?

also, we will definitely need to cut a 1.3.1 with this. good catch.

Reply:
The gogocodec hack for etcd can be removed once it's upgraded to grpc >v1.38 is released (tentatively next release from v3.5.1). We're stuck with this for a bit I'm afraid.

Reply:
FWIW, it's updated in [`main`](https://github.com/etcd-io/etcd/blob/main/api/go.mod#L11), but still unreleased